### DISCUSSION TITLE
Memory usage high!

### DISCUSSION THREAD

**Discussion by bowenxuuu** [lapce/floem#690] [Q&A]
I wrote a CounterAPP by referring to the examples on the official website of floem. However, the memory usage during running is extremely high, approximately 180MB.
I'm not sure if this is correct ?

```
#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release

use floem::{
    prelude::create_rw_signal,
    views::{button, dyn_view, Decorators},
    IntoView,
};

fn app_view() -> impl IntoView {
    let mut counter = create_rw_signal(0);
    (
        dyn_view(move || format!("Value: {}", counter)),
        (
            button("Increment").action(move || counter += 1),
            button("Decrement").action(move || counter -= 1),
        )
            .style(|s| s.flex_row().gap(6).items_center()),
    )
        .style(|s| s.flex_col().gap(6).justify_center())
}

fn main() {
    floem::launch(app_view);
}
```

```
[package]
name = "floem_learn"
version = "0.1.0"
edition = "2021"

[dependencies]
floem = "0.2.0"

[profile.release]
lto = true
opt-level = 3
panic = "abort"
codegen-units = 1

```

**Comment by jrmoulton** [comment]
This is pretty typical although it depends on a few factors. The higher the resolution of the display being used or the larger the app window is, the more memory will be used.

the CPU renderer uses significantly less memory. 

There is room for improvement in the vger renderer although long term the plan is to switch to vello as the GPU renderer. Currently, one of the reasons it isn't the primary renderer is because it uses significantly more memory than vger.
